lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+                                dimension: 50,
+                                child: CircularProgressIndicator(),
+                              ),
+                            );
+                          return _EntriesSlot(snapshot);
+                        },
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class CenterText extends StatelessWidget {
+  final DateTime day;
+  late Color _color;
+  CenterText(this._color, this.day);
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Text(
+        '${day.day}',
+        style: TextStyle(
+            fontFamily: 'assets\fonts\Barlow\Barlow-Bold.ttf',
+            color: _color,
+            fontSize: 17,
+            fontWeight: FontWeight.w700),
+      ),
+    );
+  }
+}
+
+class HighLightOpen extends StatelessWidget {
+  const HighLightOpen({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 10, bottom: 10, left: 6),
+      decoration: BoxDecoration(
+          color: Color.fromRGBO(110, 176, 230, 0.15),
+          borderRadius: BorderRadius.only(
+            bottomLeft: Radius.circular(40),
+            topLeft: Radius.circular(40),
+          )),
+    );
+  }
+}
+
+class HighLightClose extends StatelessWidget {
+  const HighLightClose({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 10, bottom: 10, right: 6),
+      decoration: BoxDecoration(
+          color: Color.fromRGBO(110, 176, 230, 0.15),
+          borderRadius: BorderRadius.only(
+            bottomRight: Radius.circular(40),
+            topRight: Radius.circular(40),
+          )),
+    );
+  }
+}
+
+class HighLightMiddle extends StatelessWidget {
+  const HighLightMiddle({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 10, bottom: 10),
+      decoration: BoxDecoration(
+        color: Color.fromRGBO(110, 176, 230, 0.15),
+      ),
+    );
+  }
+}
+
+class HighLightSingle extends StatelessWidget {
+  const HighLightSingle({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.symmetric(horizontal: 6, vertical: 10),
+      decoration: BoxDecoration(
+          color: Color.fromRGBO(110, 176, 230, 0.15),
+          borderRadius: BorderRadius.circular(40)),
+    );
+  }
+}
+
+class _EntriesSlot extends StatelessWidget {
+  final snapshot;
+
+  _EntriesSlot(this.snapshot);
+
+  String calculateDuration(String st, String et) {
+    Duration duration = DateTime.parse(et).difference(DateTime.parse(st));
+    String hour = duration.inHours % 24 == 0
```

Comment:
What's the background to doing this? If an item is 24h, don't we want it to show '24h'?